### ISSUE TITLE
support TYPESCRIPT VERSION: 4.2.4

### ISSUE THREAD

**Issue by ofirrifo** [typescript-eslint/typescript-eslint#3486]
- [X ] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

I'm using angular 12 and the typescript version updated to 4.2

On linting stage I start to get this warning
```
WARNING: You are currently running a version of TypeScript which is not officially supported by @typescript-eslint/typescript-estree.

You may find that it works just fine, or you may not.

SUPPORTED TYPESCRIPT VERSIONS: >=3.3.1 <4.1.0

YOUR TYPESCRIPT VERSION: 4.2.4

```

Can you please update the typescript version ?

**Comment by bradzacher** [comment]
https://github.com/typescript-eslint/typescript-eslint#supported-typescript-version

Your version is old. Update your packages.
Which is one of the tick boxes on the issue template.... Which you deleted.
